### DISCUSSION TITLE
MIDI FX format for Logic Pro

### DISCUSSION THREAD

**Discussion by DarkKnight-1008** [DISTRHO/Cardinal#477] [Ideas]
I would love for there to be a MIDI FX version of this like VCV has that natively allows to send/control MIDI in Logic Pro. Currently, I'm using the workaround of loading Blue Cat Patchworks into the MIDI FX slot which allows the hosting of any plugin. So I load Cardinal into THAT and get it to work. But it's tedious and I'm not convinced it's stable. I get weird behavior sometimes. For instance, I've done this on a track where the instrument is Pigments. I want to use Cardinal as MIDI FX to use its interesting sequencers etc. But whenever I open this project, it sets the pitchbend of the Pigments instance all the way down. Holding in that position. I have to open up Pigments and reset it every time I load the project.

**Comment by dromer** [comment]
It seems that Logic is the only DAW that has a notion of such "MIDI FX" plugin, so expect a resounding "no" to this request.

Cardinal is not going to tailor to one specific DAW, the AU is there already for convenience for Logic users (is there any other DAW that only loads AU?). For me this works in other DAWs so I'd say the problem is with Logic being weird.

**Reply by DarkKnight-1008** [reply]
Weird or not (and perhaps I agree that it is), it exists and this is the way it is and Logic Pro is one of the main DAWs. It's not some obscure piece of software with no market share. And plenty of developers do cater to it by having a MIDI FX. As I mentioned, VCV is one of them. I don't think it's ridiculous to request or at least ask about it. And for all of those reasons I don't think it deserves a "resounding no". A polite and considered no will do. But I'd still like to see it happen, so I'd also accept a tentative yes.

**Comment by falkTX** [comment]
On AU there is a separate category for MIDI FX, if I understood things correctly..
I didnt check the VCV 2.2 update, does this mean on macOS it ships with 3 variants?

**Comment by DarkKnight-1008** [comment]
Yeah it has all 3 it seems. I was watching a video on YouTube with someone demonstrating it. It had the Synth (instrument), The audio effect AND the MIDI FX version. As I say, I'm currently using a workaround to use it to manipulate MIDI, but it would be awesome to not have to do that. It is a bit annoying that Logic is the only DAW with the peculiarity as it can often mean developers don't cater to it, but I'm really taken with CArdinal and I love what some of these modules can do. I'd love to be able to use it to control things outside of Cardinal though which really opens up the whole world.

**Comment by michaelkeuter** [comment]
Hi, I also would like to see Cardinal as a MIDI FX version in Logic Pro.

**Comment by alltiagocom** [comment]
I agree 100% that a MIDI FX version would be awesome! Sending some custom envelopes and arps/sequences to other synths would be great!

**Comment by DarkKnight-1008** [comment]
Well they already added it with the latest version (which I’m really grateful for by the way).

**Comment by falkTX** [comment]
In case you missed it, this was added in 23.02 release

**Reply by alltiagocom** [reply]
Thanks @DarkKnight-1008 and @falkTX 
This is awesome news! Can't wait to try it!

**Reply by charlieperrone** [reply]
Has it been added? I am on v25 and I do not see it as an option in MidiFX. I'm in Logic Pro X

**Reply by alltiagocom** [reply]
> Has it been added? I am on v25 and I do not see it as an option in MidiFX. I'm in Logic Pro X

I'm on v23.10 and I have it:

<img width="515" height="246" alt="image" src="https://github.com/user-attachments/assets/95f27118-b58c-46f8-b6a1-5840ba34874d" />

Just in case, make sure it's not unchecked in Plugin Manager:

<img width="464" height="90" alt="image" src="https://github.com/user-attachments/assets/79be1351-31f6-41f7-a8b2-60183e9bf662" />

**Reply by charlieperrone** [reply]
Hm, I just downloaded the latest release for mac os universal from here. https://github.com/DISTRHO/Cardinal/releases

The versions I have in my plugin manager are different as well. Where did you install from?
<img width="874" height="153" alt="Screen Shot 2025-10-16 at 12 40 39 PM" src="https://github.com/user-attachments/assets/3fccd9b1-c343-4871-9c79-590e5d9e8608" />

**Reply by charlieperrone** [reply]
<img width="588" height="251" alt="Screen Shot 2025-10-16 at 12 48 39 PM" src="https://github.com/user-attachments/assets/2e7b61df-0c84-4edf-8c72-625a50db691c" />

My cardinal native is v25, I just dont understand why the plugins are different.

Backing off to the next most recent hasn't helped either

**Reply by alltiagocom** [reply]
I downloaded my versions probably from the same place and did a normal install.
It's in fact weird that your versions say 0.25 and mine say 23.10, but maybe Filipe can clarify this?

I never used CardinalNative, because I always got this:

<img width="424" height="162" alt="image" src="https://github.com/user-attachments/assets/81e2a7b5-913f-4cbc-ad54-85b8ffe289d5" />

and to be honest, the whole "Jack" thing has always been a big confusing for me, for years! And since I don't really see any relevance for my workflow to use the stand alone, I never bothered. I also have VCV Rack so if I want to just play around, I use that.

At this point I guess only @falkTX can help you figure it out, I'm afraid. Good luck!

**Comment by michaelkeuter** [comment]
Can that be Apple Silicon vs. Intel CPU related? On my M2 I don‘t see the MIDI plugin neither in v25.06 in Logic 11. And there are only 2 Distrho plugins (instead of 3) in plugin manager.

**Comment by falkTX** [comment]
the AU MIDI version is no longer available since the switch away from JUCE into a native/direct implementation. That is why its missing.

I didnt expect so many people to request it back, I will see about adding it back in the native implementation
